src/files/files_client.py

```diff
@@ -297,58 +297,65 @@ def _extract_path_from_url(self, url: str) -> str:
         return url
 
     def list_folder(
-        self, path: str, flag: str | None = None
+        self, path: str, recursive: bool = False
     ) -> dict[str, FileMetadata]:
         """
         List files and folders at the specified path. Returned dictionary keys will have
         the full remote path, so the resulting path key can be used directly for download, etc.
         Args:
             path: Remote path where to list files from
-            flag: Optional "recursive" flag. TBD: Currently always recursive?
+            recursive: Optional "recursive" flag. If specified will include sub-folder content.
         Returns:
             Dictionary mapping file paths to FileMetadata objects, maintaining the order
             from the underlying list response. Key string will include the argument
             path part; it is not removed.
         """
         remote_path = self._extract_path_from_url(path)
+        result = {}
+        continuation_token = None
+        max_iterations = 10
+        current_iteration = 0
 
-        if flag == "recursive":
-            raise ValueError("list_folder recursive flag not implemented")
+        while current_iteration < max_iterations:
+            call_args = genapi._get_get_object_kwargs(
+                orgFriendlyId=self.organization_id,
+                filePath=remote_path,
+                list_type=2,
+                recursive="true" if recursive == True else None,
+                continuation_token=continuation_token,
```

Comment:
first, it looks like continuation_token is always None here.

second, the machinery for all of this seems to simply allow pagination...and this seems like a very unorthodox way of doing it. why can't we simply paginate responses and fetch pages as needed?